c/datasetjson.c

```diff
@@ -1671,7 +1669,117 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
   finishResponse(response);     
 #endif /* __ZOWE_OS_ZOS */
 }
+//new function
+
+static int getDSCB(char* datasetName, char* dscb, int* dscbLength){
+  Volser volser = {0};
+  DatasetName dsn = {0};
+  memset(dsn.value, ' ', 44);
+  
+  int lParenIndex = indexOf(datasetName, strlen(datasetName),'(',0);
+  
+  //String parsing operation, gets rid of of 3 initial unnecessary characters-> //'
+  if (lParenIndex > 0){
+    memcpy(dsn.value,datasetName+3,lParenIndex-3);
+  }
+  //String parsing operation, gets rid of of 3 initial unnecessary characters-> //'
+  else{
+    memcpy(dsn.value,datasetName+3,strlen(datasetName)-4);
+  }
+    
+  int volserSuccess = getVolserForDataset(&dsn, &volser);
+  
+  if(!volserSuccess){
+    
+    int rc = obtainDSCB1(dsn.value, sizeof(dsn.value),
+                           volser.value, sizeof(volser.value),
+                           dscb);
+                           
+    *dscbLength = sizeof(dscb);
+
+    if (rc == 0){
+      if (DSCB_TRACE){
+        zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_WARNING, "DSCB for %.*s found\n", sizeof(dsn.value), dsn.value);
+        dumpbuffer(dscb,INDEXED_DSCB);
+      }
+    }
+    return 0;
+  }
+  else {
+    return 1;
+  }
+}
+
+void newDatasetMember(HttpResponse* response, char* datasetPath, char* memberName) {
+  char dscb[INDEXED_DSCB] = {0};
+  int dscbLength = 0;
+  if (getDSCB(datasetPath, dscb, &dscbLength) == 0) {
+    if (isPartionedDataset(dscb)) {
+      char *overwriteParam = getQueryParam(response->request,"overwrite");
+      int overwrite = !strcmp(overwriteParam, "true") ? TRUE : FALSE;
+      char fullPath[DATASET_MEMBER_MAXLEN]; //Max dataset path length #define them
+      int apostrophePos = lastIndexOf(datasetPath, DATASET_PATH_MAX, '\''); //index of apostrophe
+      datasetPath[apostrophePos] = 0; // remove last apostrophe by null termination
+      //concatenates dataset name with member name
+      snprintf(fullPath, DATASET_MEMBER_MAXLEN + 1, "%s(%s)'", datasetPath, memberName);
+      FILE* datasetExists = fopen(fullPath,"r");
+      if (datasetExists && overwrite != TRUE) {//Member already exists and overwrite wasn't specified
+        fclose(datasetExists);
+        respondWithError(response, HTTP_STATUS_BAD_REQUEST, "Member already exists and overwrite wasn't specified");
+      }
+      else { // Member doesn't exist
+        if (datasetExists) {
+          fclose(datasetExists);
+        }
+        FILE* newMember = fopen(fullPath, "w");
+        if (!newMember){
+          respondWithError(response, HTTP_STATUS_BAD_REQUEST, "Bad dataset name");
+          return;
+        }
+        if (fclose(newMember) == 0){
+          response200WithMessage(response, "Successfully created member");
+        }
+        else {
+          respondWithError(response, HTTP_STATUS_BAD_REQUEST, "Could not close dataset");
+        }
+      }
+    }
+    else {
```

Comment:
I usually do something like:

bool isPDS = isPartionedDataset(dscb);
if (!isPDS) {
  return error
}

It's more obvious about bad conditions than finding them at the very end. That comes down to preference, but I thought I'd give my opinion on readability.